crdb/tx.go

```diff
@@ -101,6 +101,26 @@ func Execute(fn func() error) (err error) {
 	}
 }
 
+type txConfigKey struct {}
+
+// WithMaxRetries configures context so that ExecuteTx retries tx specified
+// number of times when encountering retryable errors.
+// Setting retries to 0 will retry indefinitely.
+func WithMaxRetries(ctx context.Context, retries int) context.Context {
```

Comment:
could you add a basic test to verify something like

```
ctx = WithMaxRetries(ctx, n)
require.Equal(t, n, numRetriesFromContext(ctx)
```

Reply:
Done.